src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx

```diff
@@ -48,30 +56,80 @@ export function ListAssetsEditor({ query, handleQueryChange, datasource }: Props
     handleQueryChange(updatedQuery, true);
   }
 
+  function validateTakeValue(value: number, TAKE_LIMIT: number) {
+    if (isNaN(value) || value < 0) {
+      return { message: takeErrorMessages.greaterOrEqualToZero, take: undefined };
+    }
+    if (value > TAKE_LIMIT) {
+      return { message: takeErrorMessages.lessOrEqualToTenThousand, take: undefined };
+    }
+    return { message: '', take: value };
+  }
+
+  function onTakeChange(event: React.FormEvent<HTMLInputElement>) {
+    const value = parseInt((event.target as HTMLInputElement).value, 10);
+    const { message, take } = validateTakeValue(value, TAKE_LIMIT);
+    setRecordCountInvalidMessage(message);
+    if (query.take !== take) {
+      const updatedQuery = { ...query, take };
+      handleQueryChange(updatedQuery, true);
+    }
+  };
+
+
   return (
     <>
-      <InlineField label="Output" labelWidth={22} tooltip={tooltips.listAssets.outputType}>
-        <RadioButtonGroup
-          options={outputTypeOptions}
-          value={query.outputType || OutputType.Properties}
-          onChange={onOutputTypeChange}
-        />
-      </InlineField>
-      <InlineField
-        label="Filter"
-        labelWidth={22}
-        tooltip={tooltips.listAssets.filter}
-      >
-        <AssetQueryBuilder
-          filter={query.filter}
-          workspaces={workspaces}
-          systems={systems}
-          globalVariableOptions={datasource.globalVariableOptions()}
-          areDependenciesLoaded={areDependenciesLoaded}
-          onChange={(event: any) => onParameterChange(event)}
-        ></AssetQueryBuilder>
+      <Stack direction='row'>
+        <Stack direction='column'>
+          <InlineField label="Output" labelWidth={22} tooltip={tooltips.listAssets.outputType}>
+            <RadioButtonGroup
+              options={outputTypeOptions}
+              value={query.outputType || OutputType.Properties}
+              onChange={onOutputTypeChange}
+            />
+          </InlineField>
+          <div className="workorders-horizontal-control-group">
+            <InlineField
+              label="Filter"
+              labelWidth={22}
+              tooltip={tooltips.listAssets.filter}
+            >
+              <AssetQueryBuilder
+                filter={query.filter}
+                workspaces={workspaces}
+                systems={systems}
+                globalVariableOptions={datasource.globalVariableOptions()}
+                areDependenciesLoaded={areDependenciesLoaded}
+                onChange={(event: any) => onParameterChange(event)}
```

Comment:
```suggestion
                onChange={(event: any) => onParameterChange(event)}
```
Check if type can be specified so we avoid any